ArduCopter/AP_Arming.cpp

```diff
@@ -633,7 +633,7 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
         SRV_Channels::set_emergency_stop(false);
         // if we are using motor Estop switch, it must not be in Estop position
     } else if (rc().find_channel_for_option(RC_Channel::AUX_FUNC::MOTOR_ESTOP) && SRV_Channels::get_emergency_stop()){
-        gcs().send_text(MAV_SEVERITY_CRITICAL,"Arm: Motor Emergency Stopped");
+        check_failed(true, "Motor Emergency Stopped");
```

Comment:
This should be using `display_failure`, despite what the other blocks in here think.

There's zero reason to ignore that flag - it's clear enough what the caller wants to happen.

Reply:
If we want to use `display_failure`, we have to move these checks from arm_checks to pre_arm_checks function as like other vehicles.

Plane:
https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/AP_Arming.cpp#L103
Rover:
https://github.com/ArduPilot/ardupilot/blob/master/Rover/AP_Arming.cpp#L81

This is not a small change. Can I include it in this PR?

Reply:
Ah.  My mistake, sorry.  I did actually look back but failed.

I'm not sure about moving these to prearm checks.  Probably worth bringing up on the devcall.